Fix for issue #10725: disable lp-ticker for STM targets which uses RTC/LSI for lp-ticker

mprse opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

This PR is continuation of PR https://github.com/ARMmbed/mbed-os/pull/12135 - which will be closed.

`tests-mbed_drivers-lp_timer` is failing on `UBLOX_C030_U201` (can't reproduce the failure locally).

According to the documentation of `STM32F437VG76` MCU:

The LSI RC acts as an low-power clock source that can be kept running in Stop and
Standby mode for the independent watchdog (IWDG) and Auto-wakeup unit (AWU). The
clock frequency is around 32 kHz. For more details, refer to the electrical characteristics
section of the datasheets.

It seems that typical `LSI` frequency is `32 kHz`, but it may vary from `17` to `47 kHz`!
This means that lp-timer test may fail on the same board because lp-ticker frequency is unstable.

The proposition is to disable lp-ticker for STM targets which uses `RTC/LSI` to drive lp-ticker - `LSI` is unstable and breaks accuracy requirements for low power ticker.

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

@jeromecoutant 

----------------------------------------------------------------------------------------------------------------

mprse committed:
Disable lp-ticker for STM targets which uses RTC/LSI for lp-ticker

According to the documentation of STM32F437VG76 MCU:

The LSI RC acts as an low-power clock source that can be kept running in Stop and
Standby mode for the independent watchdog (IWDG) and Auto-wakeup unit (AWU). The
clock frequency is around 32 kHz. For more details, refer to the electrical characteristics
section of the datasheets.

It seems that typical LSI frequency is 32 kHz, but it may vary from 17 to 47 kHz!
This means that lp-timer test may fail on the same board because lp-ticker frequency is unstable.

0xc0170 reviewed (approved):
Assuming there is no fallback option for these targets. @ARMmbed/team-ublox to confirm.

mprse committed:
Enable lp-ticker based on LSE for ublox C030 family.

mprse committed:
Disabled BLE feature for targets without lp-ticker

mprse committed:
MTB_UBLOX_ODIN_W2: Reenable lp-ticker and BLE

Make an exception for this target and enable lp-ticker when LSE is not available.
This exception is added because of with disabled lp-ticker (and BLE feature) we have the following linker error with ARM compiler:
[Error] @0,0: L6218E: Undefined symbol hciCmdAlloc (referred from BUILD/tests/MTB_UBLOX_ODIN_W2/ARM/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W2/sdk/TOOLCHAIN_ARM/libublox-odin-w2-driver.ar(hci_vendor_cmd.o)).
[Error] @0,0: L6218E: Undefined symbol hciCmdSend (referred from BUILD/tests/MTB_UBLOX_ODIN_W2/ARM/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W2/sdk/TOOLCHAIN_ARM/libublox-odin-w2-driver.ar(hci_vendor_cmd.o)).

It looks like the folloiwng library must be updated:
targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W2/sdk/TOOLCHAIN_ARM/libublox-odin-w2-driver.ar

adbridge reviewed (dismissed):
LGTM as far as I can tell

0xc0170 commented on targets/targets.json:
```diff
@@ -6558,17 +6543,9 @@
-            "2",
-            "5"
-        ],
+        "device_has_add": ["MPU", "FLASH"],
```
Can you rebase and fix the styling? It was updated on master

mprse replied:
Is there any way to fix the styling automatically, like `astyle` ?

adbridge replied:
We run astyle but it doesn't auto fix so you will need to do it manually for now. Though nothing stopping you running astyle locally and have it do it for you....

0xc0170 reviewed (dismissed):
Rebase needed , it reverts styling changes on master

mprse committed:
targets.json: fix styling
